### PR TITLE
tests: allow list breaching min resident size in statvfs test

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem

This test would sometimes violate the min resident size during disk eviction
and fail due to the generate warning log.

Disk usage candidate collection only takes into account active tenants.
However, the statvfs call takes into account the entire tenants directory,
which includes tenants which haven't become active yet.

After re-starting the pageserver, disk usage eviction may kick in *before*
both tenants have become active. Hence, the logic will try to satisfy the
disk usage requirements by evicting everything belonging to the active tenant,
and hence violating the tenant minimum resident size.

## Summary of changes
Allow the warning with an accompanying rationale comment.

Closes https://github.com/neondatabase/neon/issues/8267

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by VladLazar** [comment]
@problame this is kind of interesting. When restarting under disk pressure, we can be unfair to the tenants that are quicker to activate. Was this a concious trade-off?

**Comment by problame** [comment]
> When restarting under disk pressure, we can be unfair to the tenants that are quicker to activate. Was this a concious trade-off?

Not a conscious trade-off. (NB: @koivunej worked on this more recently than me.)

I think at some point I was proposing to do one special disk-usage-based eviction early during startup.
I think it was in the context of me proposing that we abort pageserver process on ENOSPC.
https://neondb.slack.com/archives/C033RQ5SPDH/p1716896265296329

With today's implementation, I suppose it would have to look like this

```
PS starts up
init_tenant_mgr spawns attach tasks
barrier: piles up all tenants before they activate
do one disk-usage-based eviction run
invariant: we are within the free space target
open barrier => tenants activate
```

(secondary mode tenants would also have to pile up on that barrier)

**Review by problame — APPROVED** [review]

**Comment by koivunej** [comment]
Hmm... I wonder if this is fallout of the activation changes, because I think last I looked those we started background jobs only after all tenants active.

> When restarting under disk pressure, we can be unfair to the tenants that are quicker to activate. Was this a concious trade-off?

I think the answer is that _before_ we would not have done disk pressure eviction until all tenants get to activate. I also don't think this is very likely situation, but even in that case the relative order should help. Sadly it is still not enabled in prod, because utilization score is missing.

**Review by koivunej — APPROVED** [review]
anyways the change is good.

File: test_runner/regress/test_disk_usage_eviction.py [hunk 1]
```diff
@@ -794,6 +794,16 @@ def less_than_max_usage_pct():
...
 
+    # Disk usage candidate collection only takes into account active tenants.
+    # However, the statvfs call takes into account the entire tenants directory,
+    # which includes tenants which haven't become active yet.
+    #
+    # After re-starting the pageserver, disk usage eviction may kick in *before*
+    # both tenants have become active. Hence, the logic will try to satisfy the
+    # disk usage requirements by evicting everything belonging to the active tenant,
+    # and hence violating the tenant minimum resident size.
+    env.neon_env.pageserver.allowed_errors.append(".*" + GLOBAL_LRU_LOG_LINE)
```
Comment (koivunej) [inline test_runner/regress/test_disk_usage_eviction.py:805]: Shouldn't this be in the `EvictionEnv.pageserver_start_with_disk_usage_eviction`?

Comment (VladLazar) [reply test_runner/regress/test_disk_usage_eviction.py:805]: I chose the smaller hammer. We can move if the rest of these tests are flaky too.

**Comment by VladLazar** [comment]
> Hmm... I wonder if this is fallout of the activation changes, because I think last I looked those we started background jobs only after all tenants active.

Yeah, that would make sense. Perhaps we should only do disk usage eviction after all the initial tenants have become active.

**Pull request merged**